pkg/render/node.go

```diff
@@ -695,6 +696,12 @@ func (c *nodeComponent) nodeVolumes() []v1.Volume {
 		},
 	}
 
+	if c.cr.CalicoNetwork != nil &&
+		c.cr.CalicoNetwork.LinuxDataplane != nil &&
+		*c.cr.CalicoNetwork.LinuxDataplane == operatorv1.LinuxDataplaneBPF {
+		volumes = append(volumes, v1.Volume{Name: "bpffs", VolumeSource: v1.VolumeSource{HostPath: &v1.HostPathVolumeSource{Path: "/sys/fs/bpf", Type: &dirMustExist}}})
```

Comment:
My idea was that, since the user is opting in to BPF, we can be more strict here (and allow progress towards a non-privileged calico-node); doing it this way should halt the roll out if the bpffs isn't mounted, which seems like a good compromise and at least gives some feedback to the user.